cloud-controller-manager/cmd/digitalocean-cloud-controller-manager-admission-server/main.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2023 DigitalOcean
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"golang.org/x/oauth2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager/signals"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	"github.com/digitalocean/digitalocean-cloud-controller-manager/cloud-controller-manager/do"
+	"github.com/digitalocean/godo"
+)
+
+const (
+	doAccessTokenEnv    string = "DO_ACCESS_TOKEN"
+	doOverrideAPIURLEnv        = "DO_OVERRIDE_URL"
+	doClusterIDEnv             = "DO_CLUSTER_ID"
+	doClusterVPCIDEnv          = "DO_CLUSTER_VPC_ID"
+)
+
+func init() {
+	log.SetLogger(zap.New())
+}
+
+type tokenSource struct {
+	AccessToken string
+}
+
+func (t *tokenSource) Token() (*oauth2.Token, error) {
+	token := &oauth2.Token{
+		AccessToken: t.AccessToken,
+	}
+	return token, nil
+}
+
+func main() {
+	if err := startAdmissionServer(); err != nil {
+		fmt.Fprintf(os.Stderr, "failed to start admission server: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func startAdmissionServer() error {
+	ll := ctrl.Log.WithName("digitalocean-cloud-controller-manager-admission-server")
+
+	server := webhook.NewServer(webhook.Options{})
+
+	godoClient, err := getGodoClient()
+	if err != nil {
+		return fmt.Errorf("getting godo client: %w", err)
+	}
+
+	lbAdmissionHandler := do.NewLBServiceAdmissionHandler(&ll, godoClient)
+	if err := lbAdmissionHandler.WithRegion(); err != nil {
+		return fmt.Errorf("injecting region into LB validator: %w", err)
```

Comment:
Appreciate if we could apply two small adjustments to the error we produce here and elsewhere in this PR:

- Start off the message with `failed to [...]` for consistency and readability reasons
- Change `%w` to `%s` to avoid extending the API surface error -- even though it wouldn't make a difference now, it could do so through a future refactoring (for a rationale, see [this section](https://go.dev/blog/go1.13-errors#whether-to-wrap) from an official Go blog post)